taxonomy-manager-rest-server/.gitignore

```diff
@@ -1,2 +1,3 @@
 /build
 /out
+/src/test/resources/privateKey.pem
```

Comment:
Why are we ignoring a fake generated private key?

Reply:
Because it's the real one. We need the real public key checked into git for use in the Application.properties, so we need to generate the test JWT using the real private key if we want to verify that it's letting the right people in

Reply:
I'm still not entirely clear on why this is being ignored - as @garyttierney mentioned, it's a fake private key (i.e. generated solely for the purpose of running tests). It's not used for anything outside of this.

Also, if we aren't checking it in, how does Jenkins execute the JWT tests without it?

Reply:
Ah, sorry - there was some face-to-face discussion on this point that didn't make it onto here. I included the key in the second commit of this PR but pre-commit has a rule of failing the build if the change contains a private key so we backed it out again. As for how to run tests end-to-end, we don't have anything that needs this yet (there's a ticket in the backlog to work out how to do it) - this change only needs that private key for running tests manually. Now that I think about it again though, I think it would probably be better to check this key in, and drop the pre-commit rule that makes it an issue, to save having to manually pass this key around

Reply:
Is it bad practice to generate the key in CI?

Reply:
@stephenwf no, and in fact @robmarch2 I think that would probably be a better idea - it feels wrong to check any key (public or private) into GitHub regardless of what it's used for. It's trivial to generate a temporary key during CI.